tests/unit/lms/services/rsa_key_service_test.py

```diff
@@ -0,0 +1,100 @@
+import json
+from unittest.mock import sentinel
+
+import pytest
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from jose import constants
+
+from lms.models import RSAKey
+from lms.services.rsa_key import RSAKeyService, factory
+from tests import factories
+
+
+class TestAESService:
+    def test_generate(self, svc, aes_service, rsa, jwk):
+        jwk.RSAKey.return_value.to_dict.return_value = {}
+
+        key = svc.generate()
+
+        rsa.generate_private_key.assert_called_once_with(
+            public_exponent=65537, key_size=2048, backend=default_backend()
+        )
+
+        # Private key
+        aes_service.build_iv.assert_called_once_with()
+        rsa.generate_private_key.return_value.private_bytes.assert_called_once_with(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PrivateFormat.TraditionalOpenSSL,
+            encryption_algorithm=svc.no_encryption,
+        )
+        aes_service.encrypt.assert_called_once_with(
+            aes_service.build_iv.return_value,
+            rsa.generate_private_key.return_value.private_bytes.return_value,
+        )
+
+        # Public key
+        public_bytes = (
+            rsa.generate_private_key.return_value.public_key.return_value.public_bytes
+        )
+        public_bytes.assert_called_once_with(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PublicFormat.SubjectPublicKeyInfo,
+        )
+        jwk.RSAKey.assert_called_once_with(
+            algorithm=constants.Algorithms.RS256,
+            key=public_bytes.return_value.decode("utf-8"),
+        )
+
+        assert key.private_key == aes_service.encrypt.return_value
+        assert key.aes_cipher_iv == aes_service.build_iv.return_value
+        assert json.loads(key.public_key)["use"] == "sig"
+
+    def test_private_key(self, aes_service, svc):
+        key = RSAKey(aes_cipher_iv=sentinel.iv, private_key=sentinel.private_key)
+
+        private_key = svc.private_key(key)
+
+        aes_service.decrypt.assert_called_once_with(sentinel.iv, sentinel.private_key)
+        assert private_key == aes_service.decrypt.return_value
+
+    @pytest.mark.usefixtures("with_keys")
+    def test_all(self, svc):
+        keys = svc.all()
+
+        assert not any(k.expired for k in keys)
```

Comment:
```suggestion
        assert not any(key.expired for key in keys)
```

Reply:
It feels a bit like we should be checking that we get the ones we expect here. For example if this returned only one, this test wouldn't fail.

I like the fixture adding some noise elements, but I think it might work just as well if it returned the non expired keys and we compared to them.